src/Microsoft.AspNet.Mvc.Core/Infrastructure/MvcRouteHandler.cs

```diff
@@ -87,7 +87,7 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
                 {
                     _logger.ExecutingAction(actionDescriptor);
 
-                    var startTime = Environment.TickCount;
+                    var startTime = DateTime.UtcNow;
```

Comment:
IMHO, avoid using `Environment.TickCount` or `DateTime.UtcNow` for profiling is a really good thing, as they don't offer enough precision for that (IIRC, `Environment.TickCount`'s resolution is between 10 and 15 ms).

`Stopwatch` is probably the best candidate for profiling.